de.metas.contracts/src/main/java/de/metas/contracts/interceptor/C_Flatrate_Term.java

```diff
@@ -569,4 +582,108 @@ public void setMasterDocumentNo(final I_C_Flatrate_Term term)
 			}
 		}
 	}
+
+	@ModelChange(timings = {
+			ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_AFTER_CHANGE
+	}, ifColumnsChanged = {
+			I_C_Flatrate_Term.COLUMNNAME_ContractStatus, I_C_Flatrate_Term.COLUMNNAME_C_FlatrateTerm_Next_ID
+	})
+	public void updateContractStatusInOrder(final I_C_Flatrate_Term term)
+	{
+		final I_C_OrderLine ol = term.getC_OrderLine_Term();
+		if (ol == null)
+		{
+			return;
+		}
+
+		final I_C_Order contractOrder = InterfaceWrapperHelper.create(ol.getC_Order(), I_C_Order.class);
+
+		if (InterfaceWrapperHelper.isNew(term) && !Services.get(IContractsDAO.class).termHasAPredecessor(term))
+		{
+			final ContractOrderService contractOrderRepository = Adempiere.getBean(ContractOrderService.class);
+			contractOrderRepository.setOrderContractStatusAndSave(contractOrder, I_C_Order.CONTRACTSTATUS_Active);
+			return;
+		}
+
+		final OrderId orderId = OrderId.ofRepoId(contractOrder.getC_Order_ID());
+		
+		final ContractOrderService contractOrderRepository = Adempiere.getBean(ContractOrderService.class);
+		final List<OrderId> orderIds = contractOrderRepository.retrieveAllContractOrderList(orderId);
+
+		updateStatusIfNeededWhenExtendind(term, orderIds);
+
+		updateStatusIfNeededWhenCancelling(term, orderIds);
+
+		updateStausIfNeededWhenVoiding(term);
+	}
+
+	private void updateStatusIfNeededWhenExtendind(final I_C_Flatrate_Term term, final List<OrderId> orderIds)
+	{
+		if (InterfaceWrapperHelper.isValueChanged(term, I_C_Flatrate_Term.COLUMNNAME_C_FlatrateTerm_Next_ID)
+				&& term.getC_FlatrateTerm_Next_ID() > 0
+				&& orderIds.size() > 1) // we set the Extended status only when an order was extended
+		{
+			final ContractOrderService contractOrderRepository = Adempiere.getBean(ContractOrderService.class);
+			// update order contract status to extended
+			final I_C_Order contractOrder = InterfaceWrapperHelper.create(term.getC_OrderLine_Term().getC_Order(), I_C_Order.class);
+			orderIds.forEach(id -> {
+				if (id.getRepoId() != contractOrder.getC_Order_ID())  // different order from the current one
+				{
+					final I_C_Order order = InterfaceWrapperHelper.load(id, I_C_Order.class);
```

Comment:
pls avoid `load` as much as possible. Use the corresponding DAO/Repository to retrieve that object.